### DISCUSSION TITLE
Disabling auto-merge if configuration changes

### DISCUSSION THREAD

**Discussion by 0xSSA** [renovatebot/renovate#21781] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

5.58.1

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

Github Enterprise

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Let's say you want to auto-merge all major versions for some time. Later, on you decide you decide to only consider minor versions. 

During the time auto-merge was set for all major versions, Renovate creates PR's for major versions and sets the PR to be auto-merged. However, if you later change the configuration to only consider minor versions and there might be some PR's still open for major versions, Renovate will still auto-merge the major versions, even though the configuration is currently set to minor, i.e. Renovate do not disable auto-merge on the open PR's for major versions when the configuration changes. 

I would expect that the auto-merge setting gets changed on all the PR's still open, that do not follow the current configuration. 

This is the auto-merge configuration I use for minor / major: 

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "packageRules": [
    {
      "matchUpdateTypes": ["major"], 
      "automerge": true,
      "automergeType": "pr",
      "platformAutomerge": true
    }
  ]
}
```

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "packageRules": [
    {
      "matchUpdateTypes": ["minor"], 
      "automerge": true,
      "automergeType": "pr",
      "platformAutomerge": true
    }
  ]
}
```

### What you tried so far.

Looked through the documentation, to see if there are any available configurations I need to set to make this work. However, no luck. 

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by viceice** [comment, marked as answer]
not possible with platform automerge

**Reply by 0xSSA** [reply]
Is it possible when platformAutomerge is not set?

**Reply by viceice** [reply]
yes, than on next run after config change renovate would no longer automerge

**Reply by viceice** [reply]
but if the platform automerge was enabled, the platform will still automerge that PR after checks succeeds. you manually need to disable the automerge on each PR

**Reply by 0xSSA** [reply]
Okay, thank you for the clarification.
